modules/api/src/main/java/org/apache/ignite/table/criteria/CriteriaQuerySource.java

```diff
@@ -36,7 +35,8 @@ public interface CriteriaQuerySource<T> {
      * @param tx Transaction to execute the query within or {@code null} to run within implicit transaction.
      * @param criteria The predicate to filter entries or {@code null} to return all entries from the underlying table.
      * @return Iterator with query results.
-     * @throws IgniteException If failed.
+     * @throws CriteriaException If failed.
+     * @throws IllegalArgumentException If the names of the provided columns are incorrect.
```

Comment:
If you think the change is correct, then should we do the same for all public API methods?Because all (?) of public API methods validates argument and could throw this exception.

I think the exception mentioning can be dropped as it brings nothing and in normal case user can't handle and shouldn't care about it.  

Also, we can't foresee all the cases, when IllegalArgumentException could be thrown. E.g. also, the exception could be thrown if condition has an argument of unexpected type.

Reply:
Removed